packages/indexer/src/indexer.test.ts

```diff
@@ -116,4 +133,341 @@ describe("Run Test", () => {
       ]
     `);
   });
+
+  it("factory mode: indexer should merge filters and restart when needed", async () => {
+    const client = new MockClient<MockFilter, MockBlock>((request, options) => {
+      const [, mainFilter] = request.filter;
```

Comment:
I don't like the single `,` because it's easy to miss. Let's name the parameter and ignore it with a leading `_`.

```suggestion
      const [_factoryFilter, mainFilter] = request.filter;
```